bns-test/test/name_preorder.test.ts

```diff
@@ -0,0 +1,130 @@
+    namespace: "blockstack",
+    version: 1,
+    salt: "0000",
+    value: 96,
+    namespaceOwner: alice,
+    nameOwner: bob,
+    priceFunction: {
+      buckets: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+      base: 1,
+      coeff: 2,
+      noVowelDiscount: 0,
+      nonAlphaDiscount: 0,
+    },
+    renewalRule: 1,
+    nameImporter: alice,
+    zonefile: "LOREM IPSUM DOLOR SIT AMET",
+  }, {
+    namespace: "id",
+    version: 1,
+    salt: "0000",
+    value: 9600,
+    namespaceOwner: alice,
+    nameOwner: bob,
+    priceFunction: {
+      buckets: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+      base: 1,
+      coeff: 2,
+      noVowelDiscount: 0,
+      nonAlphaDiscount: 0,
+    },
+    renewalRule: 1,
+    nameImporter: alice,
+    zonefile: "LOREM IPSUM DOLOR SIT AMET",
+  }];
+
+  beforeEach(async () => {
+    const allocations = [
+      { principal: alice, amount: 10_000_000_000 },
+      { principal: bob, amount: 10_000_000 },
+      { principal: charlie, amount: 10_000_000 },
+      { principal: dave, amount: 10_000_000 },
+    ]
+    const binFile = getDefaultBinaryFilePath();
+    const dbFileName = getTempFilePath();
+    provider = await NativeClarityBinProvider.create(allocations, dbFileName, binFile);
+    bns = new BNSClient(provider);
+    await bns.deployContract();
+  });
+
+  describe("Pre-ordering the name 'bob.blockstack'", () => {
+    // it("should fail if the hash of the FQN is mal-formed");
+
+    it("should fail if Bob's balance is insufficient", async () => {
+
+      await mineBlocks(bns, 20);
+
+      var receipt = await bns.namePreorder(
+        cases[0].namespace,
+        "bob",
+        cases[0].salt, 
+        20000000, { sender: cases[0].nameOwner });
+      expect(receipt.success).eq(false);
+      expect(receipt.error).include('4001');    
+    // });
+
+    // it("should succeed if Bob's balance is provisioned", async () => {
+      receipt = await bns.namePreorder(
+        cases[0].namespace,
+        "bob",
+        cases[0].salt, 
+        200, { sender: cases[0].nameOwner });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('u33');    
+    // });
+
+    // it("should fail if the same order is being re-submitted by Bob", async () => {
+      receipt = await bns.namePreorder(
+        cases[0].namespace,
+        "bob",
+        cases[0].salt, 
+        200, { sender: cases[0].nameOwner });
+      expect(receipt.success).eq(false);
+      expect(receipt.error).include('2016');    
+    // });
+
+    // it("should succeed if the same order is being re-submitted by Alice", async () => {
+      receipt = await bns.namePreorder(
+        cases[0].namespace,
+        "bob",
+        cases[0].salt, 
+        200, { sender: alice });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('u35');    
+    // });
+
+    // it("should succeed once claimability TTL expired", async () => {
+      await mineBlocks(bns, 10);
```

Comment:
I thought the claimability TTL was longer than 10 blocks?  Isn't it 144 blocks (1 day)?

Reply:
The current values are the following:
```
(define-constant NAMESPACE_PREORDER_CLAIMABILITY_TTL u10)
(define-constant NAMESPACE_LAUNCHABILITY_TTL u10)
(define-constant NAME_PREORDER_CLAIMABILITY_TTL u10)
(define-constant NAME_GRACE_PERIOD_DURATION u5)
```

Will do a pass on this post Dec. 15th - issue tracked here https://github.com/blockstack/stacks-blockchain/issues/2083